metpy/calc/basic.py

```diff
@@ -261,36 +264,78 @@ def heat_index(temperature, rh, mask_undefined=True):
     ----------------
     mask_undefined : bool, optional
         A flag indicating whether a masked array should be returned with
-        values where heat index is undefined masked.  These are values where
-        the temperature < 80F or relative humidity < 40 percent. Defaults
-        to `True`.
+        values masked where the temperature < 80F. Defaults to `True`.
 
     See Also
     --------
     windchill
 
     """
+    temperature = atleast_1d(temperature)
+    rh = atleast_1d(rh)
+    # assign units to rh if they currently are not present
+    if not hasattr(rh, 'units'):
+        rh = rh * units.dimensionless
     delta = temperature.to(units.degF) - 0. * units.degF
     rh2 = rh * rh
     delta2 = delta * delta
 
-    # Calculate the Heat Index -- constants converted for RH in [0, 1]
-    hi = (-42.379 * units.degF
-          + 2.04901523 * delta
-          + 1014.333127 * units.delta_degF * rh
-          - 22.475541 * delta * rh
-          - 6.83783e-3 / units.delta_degF * delta2
-          - 5.481717e2 * units.delta_degF * rh2
-          + 1.22874e-1 / units.delta_degF * delta2 * rh
-          + 8.5282 * delta * rh2
-          - 1.99e-2 / units.delta_degF * delta2 * rh2)
+    # Simplifed Heat Index -- constants converted for RH in [0, 1]
+    a = -10.3 * units.degF + 1.1 * delta + 4.7 * units.delta_degF * rh
+
+    # More refined Heat Index -- constants converted for RH in [0, 1]
+    b = (-42.379 * units.degF
```

Comment:
I'd love if we could get to the point where we're only calculating `b` for the points that will use it, but that would make the code a lot more convoluted. Probably only worth considering once we have a use case where current performance is insufficient.